src/dataflow/src/render/join/mod.rs

```diff
@@ -43,7 +43,7 @@ pub use linear_join::LinearJoinPlan;
 /// A complete enumeration of possible join plans to render.
 #[derive(Clone, Debug, Serialize, Deserialize)]
 pub enum JoinPlan {
-    Linear(LinearJoinPlan),
+    Linear(Box<LinearJoinPlan>),
     Delta(DeltaJoinPlan),
```

Comment:
Doing both makes sense to me. At least, Delta join plans are weirder and more complicated, and it is unclear on what basis the boxing is done (if we add a new variant, ... what should be done?).

Reply:
My understanding is that the decision is made by the `variant_size_differences` lint at the root of the dataflow crate. The `variant_size_differences` lint is ok with `DeltaJoinPlan` for now because `DeltaJoinPlan` is currently a struct with a single `Vec`, and the objects in the `Vec` are already on the heap.

Reply:
I agree to that it'd be more intuitive to box the delta plan as well, but we could decide that we'd just use the `variant_size_differences` warning as an indicator to push members to the heap.